griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -76,6 +76,13 @@ def _base_params(self, prompt_stack: PromptStack) -> dict:
             "messages": messages
         }
 
+        # A max_tokens parameter is not required, but if it is specified by the caller, bound it to
+        # the maximum value as determined by the tokenizer and pass it to the API.
+        if self.max_tokens:
+            params["max_tokens"] = self.max_output_tokens(messages)
```

Comment:
I think this is fine but could potentially be confusing when the user sets tokens explicitly and we still sometimes override it with fewer tokens. Let's keep it this way, as this feels like a better UX, but may be document it elsewhere.